dlib/CMakeLists.txt

```diff
@@ -1,1039 +1,421 @@
+find_package(CBLAS)
+if (PKG_CONFIG_FOUND)
+   pkg_check_modules(FFMPEG IMPORTED_TARGET libavdevice libavfilter libavformat libavcodec libavutil libswresample libswscale)
 endif()
 
+# Options
+option(DLIB_ISO_CPP_ONLY         "Build using standard c++ only"  OFF)
+option(DLIB_ENABLE_ASSERTS       "Enable Debug assert statements" OFF)
+option(DLIB_ENABLE_STACK_TRACE   "Enabled Debug stack trace"      OFF)
+option(DLIB_JPEG_SUPPORT         "Enable JPEG support"            ON)
+option(DLIB_PNG_SUPPORT          "Enable PNG support"             ON)
+cmake_dependent_option(DLIB_USE_BLAS            "Enable BLAS support"      ON  "MKL_FOUND OR BLAS_FOUND"       OFF)
+cmake_dependent_option(DLIB_USE_LAPACK          "Enable LAPACK support"    ON  "MKL_FOUND OR LAPACK_FOUND"     OFF)
+cmake_dependent_option(DLIB_USE_AVX             "Enable AVX SIMD"          ON  "AVX_FOUND"                     OFF)
+cmake_dependent_option(DLIB_USE_SSE             "Enable SSE SIMD"          ON  "SSE_FOUND"                     OFF)
+cmake_dependent_option(DLIB_USE_NEON            "Enable NEON SIMD"         ON  "NEON_FOUND"                    OFF)
+cmake_dependent_option(DLIB_GIF_SUPPORT         "Enable GIF support"       ON  "GIF_FOUND"                     OFF)
+cmake_dependent_option(DLIB_WEBP_SUPPORT        "Enable WEBP support"      ON  "WEBP_FOUND"                    OFF)
+cmake_dependent_option(DLIB_NO_GUI_SUPPORT      "Disable GUI support"      OFF "X11_FOUND OR WIN32"            ON)
+cmake_dependent_option(DLIB_LINK_WITH_SQLITE3   "Enable Sqlite3 support"   ON  "SQLite3_FOUND"                 OFF)
+cmake_dependent_option(DLIB_USE_FFMPEG          "Enable FFMPEG support"    ON  "FFMPEG_FOUND"                  OFF)
+cmake_dependent_option(DLIB_USE_CUDA            "Enable CUDA support"      ON  "CUDAToolkit_FOUND;CUDNN_FOUND" OFF)
+cmake_dependent_option(DLIB_USE_MKL_SEQUENTIAL  "Enable MKL sequential"    ON  "MKL_FOUND"                     OFF)
+cmake_dependent_option(DLIB_USE_MKL_WITH_TBB    "Enable MKL tbb"           ON  "MKL_FOUND"                     OFF)
+cmake_dependent_option(DLIB_USE_MKL_WITH_GNU    "Enable MKL GNU OMP"       ON  "MKL_FOUND"                     OFF)
+
+# dlib
+add_library(dlib
+   base64/base64_kernel_1.cpp
+   bigint/bigint_kernel_1.cpp
+   bigint/bigint_kernel_2.cpp
+   bit_stream/bit_stream_kernel_1.cpp
+   entropy_decoder/entropy_decoder_kernel_1.cpp
+   entropy_decoder/entropy_decoder_kernel_2.cpp
+   entropy_encoder/entropy_encoder_kernel_1.cpp
+   entropy_encoder/entropy_encoder_kernel_2.cpp
+   md5/md5_kernel_1.cpp
+   tokenizer/tokenizer_kernel_1.cpp
+   unicode/unicode.cpp
+   test_for_odr_violations.cpp
+   fft/fft.cpp)
 
-include(cmake_utils/set_compiler_specific_options.cmake)
-
-
-# Adhere to GNU filesystem layout conventions
-include(GNUInstallDirs)
-
-if (POLICY CMP0075)
-   cmake_policy(SET CMP0075 NEW)
-endif()
-
-# default to a Release build (except if CMAKE_BUILD_TYPE is set)
-include(cmake_utils/release_build_by_default)
```

Comment:
Totally keep this too.  Tons of users expect this to be the default and if it isn't they will be surprised in a bad way.  You can't imagine the number of questions about "why is dlib so slow?" I got until I added this.

Reply:
As another example, dlib/java uses `cmake_utils/release_build_by_default` and people who use java totally don't get the whole "cmake doesn't build in release mode by default" thing.  You would be surprised at the number of users that use `cmake_utils/release_build_by_default`.

Reply:
Ok didn't realize there were java bindings. Do we want a runner for that?

Reply:
Sure, would be cool.

Reply:
Another PR